File: 4 kyu/Pyramid_Slide_Down.js
function longestSlideDown (pyramid) {
    for (var j = pyramid.length - 1; j > 0; j--) {
        var temp = [];
        for (var i = 1; i < pyramid.length - 1; i++) {
            temp.push(Math.max(pyramid[j][i -1],pyramid[j][i]));
        }
        for (var x = 0; x < temp.length; x++) {
            pyramid[j - 1][x] += temp[x]
        }    
    }
    return pyramid[0][0]
}

File: 4 kyu/Pyramid_Slide_Down.py
def longest_slide_down(pyramid):
    for j in range(len(pyramid) - 1, 0, -1):
        temp = []
        for i in range(1,len(pyramid[j])):
            temp.append(max(pyramid[j][i -1],pyramid[j][i]) )
        for x in range(len(temp)):
            pyramid[j - 1][x] += temp[x] 
    return pyramid[0][0]
